packages/keystone/fileAdapter/sberCloudFileAdapter.js

```diff
@@ -205,95 +225,134 @@ const obsRouterHandler = ({ keystone }) => {
-
-            if ((isEmpty(itemId) && isEmpty(stringItemIds)) || isEmpty(listKey)) {
-                res.status(404)
-                return res.end()
-            }
 
-            const context = await keystone.createContext({ authentication: { item: req.user, listKey: 'User' } })
+            const hasSign = typeof req.query?.sign === 'string' && req.query.sign.length > 0
+
+            // Legacy download
+            if (!hasSign) {
+                const {
+                    id: itemId,
+                    ids: stringItemIds,
+                    listkey: listKey,
+                    propertyquery: encodedPropertyQuery,
+                    propertyvalue: encodedPropertyValue,
+                } = meta
+                const propertyQuery = !isNil(encodedPropertyQuery) ? decodeURI(encodedPropertyQuery) : null
+                const propertyValue = !isNil(encodedPropertyValue) ? decodeURI(encodedPropertyValue) : null
+
+                if ((isEmpty(itemId) && isEmpty(stringItemIds)) || isEmpty(listKey)) {
+                    res.status(404)
+                    return res.end()
+                }
 
-            let hasAccessToReadFile
+                const context = await keystone.createContext({ authentication: { item: req.user, listKey: 'User' } })
 
-            // If user has access to at least one of the objects with this file => user has access to read file
-            if (!isEmpty(stringItemIds) && isString(stringItemIds)) {
-                const itemIds = stringItemIds.split(',').filter(id => UUID_REGEXP.test(id))
-                const items = await getItems({
-                    keystone,
-                    listKey,
-                    itemId,
-                    context,
-                    where: { id_in: itemIds, deletedAt: null },
-                })
+                let hasAccessToReadFile
 
-                hasAccessToReadFile = items.length > 0
-            }
+                // If user has access to at least one of the objects with this file => user has access to read file
+                if (!isEmpty(stringItemIds) && isString(stringItemIds)) {
+                    const itemIds = stringItemIds.split(',').filter(id => UUID_REGEXP.test(id))
+                    const items = await getItems({
+                        keystone,
+                        listKey,
+                        itemId,
+                        context,
+                        where: { id_in: itemIds, deletedAt: null },
+                    })
+
+                    hasAccessToReadFile = items.length > 0
+                }
+
+                if (itemId && !hasAccessToReadFile) {
+                    let returnFields = 'id'
 
-            if (itemId && !hasAccessToReadFile) {
-                let returnFields = 'id'
+                    // for checking property we have to include property name in the return fields list
+                    if (isString(propertyQuery)) {
```

Comment:
Looks like propertyQuery injection. Need to validate it

Reply:
It's original code of the adapter, so I don't think it's related to this PR